sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -984,37 +1040,47 @@ case class GpuParquetMultiFilePartitionReaderFactory(
-      } catch {
-        case e: FileNotFoundException if ignoreMissingFiles =>
-          logWarning(s"Skipped missing file: ${file.filePath}", e)
-          ParquetFileInfoWithBlockMeta(new Path(new URI(file.filePath)), Seq.empty,
-            file.partitionValues, null, null, false, false, false)
-        // Throw FileNotFoundException even if `ignoreCorruptFiles` is true
-        case e: FileNotFoundException if !ignoreMissingFiles => throw e
-        // If ignoreMissingFiles=true, this case will never be reached. But it's ok
-        // to leave this branch here.
-        case e@(_: RuntimeException | _: IOException) if ignoreCorruptFiles =>
-          logWarning(
-            s"Skipped the rest of the content in the corrupted file: ${file.filePath}", e)
-          ParquetFileInfoWithBlockMeta(new Path(new URI(file.filePath)), Seq.empty,
-            file.partitionValues, null, null, false, false, false)
-      }
+    val startTime = System.nanoTime()
+
+    def addBlockMeta(metaAndFile: BlockMetaWithPartFile): Unit = {
+      val singleFileInfo = metaAndFile.meta
       clippedBlocks ++= singleFileInfo.blocks.map(block =>
         ParquetSingleDataBlockMeta(
           singleFileInfo.filePath,
           ParquetDataBlock(block),
-          file.partitionValues,
+          metaAndFile.file.partitionValues,
           ParquetSchemaWrapper(singleFileInfo.schema),
           singleFileInfo.readSchema,
           new ParquetExtraInfo(singleFileInfo.isCorrectedRebaseMode,
-            singleFileInfo.isCorrectedInt96RebaseMode, singleFileInfo.hasInt96Timestamps)))
+            singleFileInfo.isCorrectedInt96RebaseMode,
+            singleFileInfo.hasInt96Timestamps)))
+    }
+
+    if (numFilesFilterParallel > 0) {
+      val tc = TaskContext.get()
+      val tasks = new java.util.ArrayList[Future[Array[BlockMetaWithPartFile]]]()
+      logDebug("Using parallel threads to do the filter with the Coalescing reader, " +
+        s"$numFilesFilterParallel files per thread")
+      files.sliding(numFilesFilterParallel, numFilesFilterParallel).foreach { fileGroup =>
+        val threadPool = MultiFileReaderThreadPool.getOrCreateThreadPool(numThreads)
+        tasks.add(threadPool.submit(
+          new CoalescingFilterRunner(footerReadType, tc, fileGroup, conf, filters, readDataSchema)))
+      }
+      for (future <- tasks.asScala) {
+        val fileBlockMetaArray = future.get()
+        fileBlockMetaArray.foreach { metaAndFile =>
+          addBlockMeta(metaAndFile)
+        }
+      }
+    } else {
+      files.map { file =>
+        val metaAndFile = filterBlocksForCoalescingReader(footerReadType, file, conf,
+          filters, readDataSchema)
+        addBlockMeta(metaAndFile)
+      }
     }
     metrics.get("scanTime").foreach {
-      _ += TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - currentTime)
+      _ += TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime)
```

Comment:
Isn't this also effectively the filter time if we want to always report it as wall-clock rather than sometimes one way, sometimes the other?

Reply:
yes I was just putting it off to do under the other issue, I can do it here

Reply:
went ahead and changed this. The multi-threaded reader time for filter includes buffer as well as the comment in code states. Also added the orc and avro tracking of filter time here as well